docs/ref/contrib/admin/index.txt

```diff
@@ -1812,6 +1812,19 @@ templates used by the :class:`ModelAdmin` views:
     This uses the ``HttpRequest`` instance to filter the ``Car`` foreign key
     field to only display the cars owned by the ``User`` instance.
 
+    For more complex filters, you can use ``ModelForm.__init__()`` method to
+    filter based on an ``instance`` of your model (see
+    :ref:`fields-which-handle-relationships`). For example::
+
+        class CountryAdminForm(forms.ModelForm):
+            def __init__(self, *args, **kwargs):
+                super().__init__(*args, **kwargs)
+                if self.instance:
```

Comment:
`self.instance` always exists. (`super().__init__()` will create a blank model instance if one is not provided.)

Reply:
@carltongibson good call, I've changed it :)

I was testing how relationships would work for blank model instances and they return an empty `QuerySet`, so in theory, in most of the cases that would be the expected result right? Filtered results (in case of instance is not blank), or empty results.

Perhaps we could change this example to the following?

```python
self.fields['capital'].queryset = self.instance.cities.all()
```

Reply:
Depends 😀

If I'm only using this with existing instances then I `assert` that the instance is given, since it's a programming error if not, but that distracts from the example. 

Urgh! — it doesn't have to be 100% bullet-proof: it's illustrative.

Reply:
@carltongibson yeah no worries at all. thank you :)

Reply:
Yes `if self.instance` is not necessary anymore because `self.instance.cities.all()` will work for an empty object, so
```python
            def __init__(self, *args, **kwargs):
                super().__init__(*args, **kwargs)
                self.fields['capital'].queryset = self.instance.cities.all()
```
works as expected. Previous version (`self.instance.owner.garage_set.all()`) raised an error for an empty object.

Reply:
@felixxm I've removed the condition since I think it makes the example more concise :)